### DISCUSSION TITLE
issue: Loose type parameter in watch from UseFormMethods

### DISCUSSION THREAD

**Discussion by adamsuskin** [react-hook-form/react-hook-form#10093] [Q&A]
### Version Number

6.15.8

### Codesandbox/Expo snack

https://codesandbox.io/s/nameless-waterfall-pvk6qf?file=/src/App.tsx

### Steps to reproduce

Try to use `watch` form method and get auto-complete on the field name without explicitly re-annotating the type parameter.

### Expected behaviour

Commented in the sandbox link what I'd expect. I think this can be fixed by adding a type constraint on `TFieldName` to extend `keyof TFieldValues` but interested what the purpose of the loose `string` type is. If it's for nesting, I wonder if we could use something like `DeepKeys`.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
thanks for the issue report, v6 is 2 years behind please consider to upgrade to v7.

> const tightlyWatchedFieldArray = watch<keyof FormValues>(["firstName"]);

`watch` doesn't support generic.

> const looselyWatchedField = watch("any string");

Don't support such usage.

**Reply by adamsuskin** [reply]
Gotcha. Yeah upgrading seems fair given how old v6 is, just a shame since this is an enterprise app that will take quite a bit of work to upgrade as the resolution.

Thanks for the speedy reply!
